cilium/CFP-42466-change-cilium-pod-ip-pool-crd.md

```diff
@@ -0,0 +1,102 @@
+# CFP-42466: change-multi-ip-pool-crd
+
+**SIG: SIG-NAME** ([View all current SIGs](https://docs.cilium.io/en/stable/community/community/#all-sigs))
+
+**Begin Design Discussion:** 2025-11-04
+
+**Cilium Release:** 1.19
+
+**Authors:** kyounghoonJang <matkimchi_@naver.com>
+
+**Status:** Draft
+
+**Reference** this CFP was translated from Korean to English using AI.
+
+## Summary
+
+The current CiliumPodIPPool CRD represents CIDRs as simple string lists, preventing granular control over allocation.
+This proposal introduces a schema enhancement that supports marking specific CIDRs as non-allocatable — addressing issues like orphaned IPs and enabling gradual, safe IP pool decommissioning.
+
+## Motivation
+
+When using Multi-Pool IPAM, operators sometimes need to decommission or replace existing CIDRs.
+The only current way to stop IP allocation from a CIDR is to remove it from the pool entirely, which can cause existing pod IPs to become orphaned and trigger warnings in the operator.
+
+To support a two-step migration workflow, we propose enabling CIDRs to be marked as “non-allocatable”:
+
+1. Add a new CIDR and mark the old CIDR as non-allocatable.
+
+2. Drain and migrate workloads to nodes using the new CIDR.
+
+3. Once the old CIDR is no longer in use, remove it from the pool.
+
+This allows for controlled and predictable transitions between IP pools without service disruption.
+
+## Goals
+Introduce a new `unallocatable` flag in the `CiliumPodIPPool` CRD to mark specific CIDRs as non-allocatable.
+
+## Proposal
+
+### Overview
+
+Current structure:
+```yaml
+apiVersion: cilium.io/v2alpha1
+kind: CiliumPodIPPool
+metadata:
+  name: green-pool
+spec:
+  ipv4:
+    cidrs:
+      - 10.20.0.0/16
+      - 10.30.0.0/16
+    maskSize: 24
+```
+Proposed structure:
+```yaml
+apiVersion: cilium.io/v2alpha1
+kind: CiliumPodIPPool
+metadata:
+  name: green-pool
+spec:
+  ipv4:
+    cidrs:
+      - cidr: 10.20.0.0/16
+        unallocatable: true
+      - cidr: 10.30.0.0/16
+    maskSize: 24
+```
+
+### Implementation Details
+- Create a struct to hold CIDR and the Unallocatable Flag.
+- Implement logic in the allocator to disallow pool allocation when the Unallocatable flag is set.
+
+## Impacts / Key Questions
+
+### Impact 1 : Breaking Change - Schema Migration
+
+Since the schema has changed from storing CIDRs as simple strings in a slice to using a struct that contains both the CIDR and the Unallocatable flag, existing v2alpha users will encounter a schema mismatch issue.
+
+### Key Question: A way to apply this feature without affecting existing users
```

Comment:
nit: I'm a bit confused by the headline structure here.

If this belongs to the `Breaking Change - Schema Migration` section, it should likely be `#### Question: A way to apply this feature without affecting existing users` or similar.